pyvo/registry/rtcons.py

```diff
@@ -0,0 +1,820 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Constraints for doing registry searches.
+
+The Constraint class encapsulates a query fragment in a RegTAP query, e.g., a
+keyword, a sky location, an author name, a class of services.  They are used
+either directly as arguments to registry.search, or by passing keyword
+arguments into registry.search.  The mapping from keyword arguments to
+constraint classes happens through the _keyword attribute in Constraint-derived
+classes.
+"""
+
+import datetime
+
+from astropy import units
+from astropy import constants
+from astropy.coordinates import SkyCoord
+import numpy
+
+from ..dal import tap
+from ..dal import query as dalq
+from ..utils import vocabularies
+from .import regtap
+
+
```

Comment:
please add a `__all__` line to list the public API from this module, that would keep the namespace clear and the sphinx API documentation build happy.